drivers/sensor/mchp_tach_xec/tach_mchp_xec.c

```diff
@@ -99,9 +97,28 @@ static int tach_xec_channel_get(const struct device *dev,
 	return 0;
 }
 
+static void tach_xec_sleep_clr(const struct device *dev)
+{
+	const struct tach_xec_config * const cfg = dev->config;
+	struct pcr_regs * const pcr = (struct pcr_regs * const)(
+		DT_REG_ADDR_BY_IDX(DT_NODELABEL(pcr), 0));
```

Comment:
If your device depends on another, the best thing is to specify the dependency explicitely in Devicetree. For example, by doing:


```
&tach {
   ...
    pcr = <&pcr>;
};
```

and then collect it's address (or instance if there's a proper driver) in the config. You could then get the address by doing something like

```
.pcr_regs = (struct pcr_regs *)DT_REG_ADDR(DT_INST_PHANDLE(n, pcr));
```

However, it looks like PCR is a clock/reset controller, for which Zephyr has a better way to solve the problem (a reset API was recently introduced for example). For clocks, let's take for example STM32:

```
spi3: ... {
    ...    
    clocks = <&rcc STM32_CLOCK_BUS_APB1 0x0...>;
};
```

As you can see, we have a clock cell that specifies both the controller (pcr in your case, rcc for STM32) and then clock properties for that particular peripheral.

@erwango could you provide some feedback here? Should we continue accepting custom stuff like this when we have a better way?

Reply:
> @erwango could you provide some feedback here? Should we continue accepting custom stuff like this when we have a better way?

Tbh, on STM32 we're using same shortcut for clocks as clock_controller is a singleton.
This being said, I agree that at least the bindings should be "canonical": <&phandle cell1 cell2>;

Reply:
We will consider this for future drivers, don't want to change at this point.

Reply:
We don't need to block this PR on this point, but current drivers do need to be changed too.